### PR TITLE
ENYO-6165: Fix horizontal scroll behavior in Scroller

### PR DESCRIPTION
**Pull request by germboy**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
When using a `Scroller` that is expected to be scrolled vertically and horizontally - using `direction="both"`, the `Scroller` cannot scroll horizontally when navigating contents via 5way left/right.

The issue stems from how `Scroller` is checking its `direction` prop and calculating the next scroll position because of it. It's only accounting for the values of `'vertical'` and/or `'horizontal'`. If `'both'` is used instead, it defaults to the behavior of `'vertical'`.

### Resolution
Check for all valid values of `direction` and calculate the next scroll position accordingly.

### CODE REVIEW

**Comment by ybsung** [comment]
LGTM

File: packages/moonstone/Scroller/Scroller.js [hunk 1]
```diff
@@ -240,36 +280,24 @@ class ScrollerBase extends Component {
...
-
-			if (newItemLeft + itemWidth > (clientWidth + currentScrollLeft) && itemWidth < clientWidth) {
-				// If focus is moved to an element outside of view area (to the right), scroller will move
-				// to the right just enough to show the current `focusedItem`. This does not apply to
-				// `focusedItem` that has a width that is bigger than `this.scrollBounds.clientWidth`.
-				this.uiRefCurrent.scrollPos.left += rtlDirection * ((newItemLeft + itemWidth) - (clientWidth + currentScrollLeft));
-			} else if (newItemLeft < currentScrollLeft) {
-				// If focus is outside of the view area to the left, move scroller to the left accordingly.
-				this.uiRefCurrent.scrollPos.left += rtlDirection * (newItemLeft - currentScrollLeft);
+		if (!contains(containerRect, itemRect)) {
```
Comment (0x64) [inline packages/moonstone/Scroller/Scroller.js]: Is this condition is necessary? It looks performing some redundant comparison with following two `if` statements.

File: packages/moonstone/Scrollable/Scrollable.js [hunk 2]
```diff
@@ -439,16 +439,27 @@ class ScrollableBase extends Component {
 		const
 			spotItem = Spotlight.getCurrent(),
 			positionFn = this.childRef.current.calculatePositionOnFocus,
-			{containerRef} = this.uiRef.current.childRefCurrent;
+			{containerRef} = this.uiRef.current.childRefCurrent,
+			containerNode = containerRef.current;
```
Comment (0x64) [inline packages/moonstone/Scrollable/Scrollable.js]: This is the only one line in this function to use `containerRef`. :)

File: packages/moonstone/Scrollable/ScrollableNative.js [hunk 3]
```diff
@@ -509,7 +509,8 @@ class ScrollableBaseNative extends Component {
 		const
 			spotItem = Spotlight.getCurrent(),
 			positionFn = this.childRef.current.calculatePositionOnFocus,
-			{containerRef} = this.uiRef.current.childRefCurrent;
+			{containerRef} = this.uiRef.current.childRefCurrent,
+			containerNode = containerRef.current;
 
 		if (spotItem && positionFn && containerRef.current && containerRef.current.contains(spotItem)) {
```
Comment (0x64) [inline packages/moonstone/Scrollable/ScrollableNative.js]: We can use `containerNode` here.

File: packages/moonstone/Scroller/Scroller.js (hunk 1 above)
Comment (germboy) [reply packages/moonstone/Scroller/Scroller.js]: Good catch. That was a redundant guard based on my initial implementation.

**Pull request merged**